### PR TITLE
Remove httpcache

### PR DESCRIPTION
**Pull request by skitt**
httpcache doesn’t stream downloads, which results in huge memory
requirements when images need to be retrieved (> 20 GiB with one of
the recent rhcos images).

Since we do our own ETags-based caching, I don’t think httpcache adds
enough value to outweigh the memory requirements.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Comment by skitt** [comment]
I was going for #687 too but this feels like it should be its own PR, and the libvirt integration is a lot more work.

**Comment by cgwalters** [comment]
/approve
and LGTM.

File: pkg/tfvars/libvirt/cache.go [hunk 1]
```diff
@@ -45,12 +43,7 @@ func cachedImage(uri string) (string, error) {
 		return uri, err
 	}
 
-	cache := diskcache.NewWithDiskv(diskv.New(diskv.Options{
-		BasePath:     httpCacheDir,
```
Comment (cgwalters) [inline pkg/tfvars/libvirt/cache.go]: Optional extra: Detect when the old directory exists and warn that we don't use it anymore.

Comment (skitt) [reply pkg/tfvars/libvirt/cache.go]: > Optional extra: Detect when the old directory exists and warn that we don't use it anymore.

Ah yes, and in fact we’re still creating it here even though we don’t need it...

**Comment by abhinavdahiya** [comment]
Update the commit title to match https://github.com/openshift/installer/blob/master/CONTRIBUTING.md#commit-message-format

Also looks like it's missing dropping httpcache from vendor.

**Comment by skitt** [comment]
> Update the commit title to match https://github.com/openshift/installer/blob/master/CONTRIBUTING.md#commit-message-format

Will do.

> Also looks like it's missing dropping httpcache from vendor.

I had done that initially, but it turns out `vendor/k8s.io/client-go/discovery/round_tripper.go` also needs `httpcache`, so the build fails without it.

File: pkg/tfvars/libvirt/cache.go [hunk 2]
```diff
@@ -40,17 +38,14 @@ func cachedImage(uri string) (string, error) {
...
 	cacheDir := filepath.Join(baseCacheDir, "openshift-install", "libvirt")
 	httpCacheDir := filepath.Join(cacheDir, "http")
-	err := os.MkdirAll(httpCacheDir, 0777)
-	if err != nil {
-		return uri, err
+	// We used to use httpCacheDir, warn if it still exists since the user may
+	// want to delete it
+	_, err := os.Stat(httpCacheDir)
+	if err == nil || !os.IsNotExist(err) {
+		logrus.Infof("The installer no longer uses %v, it can be deleted", httpCacheDir)
```
Comment (abhinavdahiya) [inline pkg/tfvars/libvirt/cache.go]: `%q`

**Comment by skitt** [comment]
> @skitt: you cannot LGTM your own PR.

Thankfully ;-). @zeenix this needs a LGTM label too, can you do that or should I assign wking?

**Pull request merged**